### PR TITLE
Restore instance tests to #5172

### PR DESCRIPTION
**Pull request by andrewjstone**
Restore instance tests added in #4691 so that they work with the latest updates in #5172. The tests now use the new `StorageManagerTestHarness` from #5172 instead of the since removed `FakeStorageManager`. Additionally, the dns server code from `sled-agent/src/fakes/nexus` is used rather than the reimplementation in #4691. This shrinks the amount of code added with these tests.

### CODE REVIEW

File: sled-agent/src/instance.rs [hunk 1]
```diff
@@ -1519,3 +1519,635 @@ impl InstanceRunner {
...
+
+        let client = propolis_client::Client::new(&format!(
+            "http://{}",
+            srv.local_addr()
+        ));
+
+        (srv, client)
+    }
+
+    // make a FakeStorageManager with a "U2" upserted
```
Comment (andrewjstone) [inline sled-agent/src/instance.rs:1692]: Need to remove this stale comment

File: sled-agent/src/instance.rs [hunk 2]
```diff
@@ -1519,3 +1519,634 @@ impl InstanceRunner {
         out
     }
 }
+
+#[cfg(test)]
+mod tests {
```
Comment (smklein) [inline sled-agent/src/instance.rs:1524]: As discussed in chat - this should be illumos-only if it's using the `StorageManagerTestHarness`

**Review by smklein — APPROVED** [review]
Thank you for restoring this! I'm thankful to not need to revive the FakeStorageManager

File: sled-agent/src/instance.rs (hunk 2 above)
Comment (andrewjstone) [reply sled-agent/src/instance.rs:1524]: Done in [b7c1d67](https://github.com/oxidecomputer/omicron/pull/5294/commits/b7c1d67c7c215d6e231a2728e4d7b049e5706637)

**Pull request merged**